modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -192,23 +178,23 @@ private String getVersion(Extension extension)
         return versionResolver.resolveVersion(extension);
     }
 
-    private void declareConfigurations(ExtensionDeclarer declaration, ExtensionType<?> extensionType)
+    private void declareConfigurations(ExtensionDeclarer declaration, ExtensionElement extensionElement)
     {
-        List<TypeBasedComponent<?>> configurationClasses = extensionType.getConfigurations();
-        if (configurationClasses.isEmpty())
+        List<ConfigurationElement> configurations = extensionElement.getConfigurations();
+        if (configurations.isEmpty())
         {
-            declareConfiguration(declaration, extensionType, extensionType);
+            declareConfiguration(declaration, extensionElement, extensionElement);
         }
         else
         {
-            for (TypeBasedComponent<?> configurationClass : configurationClasses)
+            for (ConfigurationElement configuration : configurations)
             {
-                declareConfiguration(declaration, extensionType, configurationClass);
+                declareConfiguration(declaration, extensionElement, configuration);
             }
         }
     }
 
-    private void declareConfiguration(ExtensionDeclarer declaration, TypeBasedComponent<?> extensionType, TypeBasedComponent<?> configurationType)
+    private void declareConfiguration(ExtensionDeclarer declaration, ConfigurationElement extensionType, ConfigurationElement configurationType)
```

Comment:
An `ExtensionElement` extends `ConfigurationElement` I just downcasting the first element